vscode_extension/src/aacExecutableWrapper.ts

```diff
@@ -0,0 +1,155 @@
+import { execShell } from "./helpers";
+import { getOutputChannel } from "./outputChannel";
+import { window, QuickPickOptions, InputBoxOptions, OpenDialogOptions, Uri } from 'vscode';
+
+const outputChannel = getOutputChannel();
+
+interface CommandArgument {
+    name: string;
+    description: string;
+    userResponse?: string;
+}
+
+/**
+ * Prompts the user for input then executes the AaC plugin with user-provided
+ * information and arguments.
+ */
+ export async function executeAacCommand(): Promise<void> {
+
+    let availableCommands: string[] = await getAacCommandNames();
+
+    const quickPickOptions: QuickPickOptions = {
+        canPickMany: false,
+    };
+
+    window.showQuickPick(availableCommands, quickPickOptions).then(commandName => {
+        if(commandName) {
+            getAacCommandArgs(commandName).then(async (commandArguments) => {
+                await getCommandArgUserInput(commandArguments);
+                execAacShellCommand(commandName, commandArguments).then(output => {
+                    outputChannel.appendLine(output);
+                    outputChannel.show();
+                });
+            });
+        }
+    });
+}
+
+/**
+ * Gets the version of the currently installed AaC package
+ * @returns string of the version number or null if not installed.
+ */
+export async function getAaCVersion(): Promise<string|null> {
+
+    let aacVersionOutput: string = "";
+    try {
+        aacVersionOutput = await execAacShellCommand("version");
+    } catch (exception) {
+        // TODO: Change to logging or alternative
+    }
+
+    const regExp = /([0-9]+\.*){3}/;
```

Comment:
Per our discussion, this should allow the use of development versions inside virtual environments.

```suggestion
    const regExp = /([0-9]+\.*){3}|-e git.*egg=aac/;
```

Reply:
Not applicable here, the source string is output from the version command.